Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Confirm.hs

```diff
@@ -120,7 +121,7 @@ handler transporter req quote = do
             QBL.updateAddress booking.toLocation.id req.toAddress
             whenJust req.mbRiderName $ QRB.updateRiderName booking.id
             QRide.create ride
-            QDFS.updateStatus driver.id DDFS.RIDE_ASSIGNED {rideId = ride.id}
+            unless (DDFS.isPaymentOverdue driverStatus) $ QDFS.updateStatus driver.id DDFS.RIDE_ASSIGNED {rideId = ride.id}
```

Comment:
It's better to move this check in `updateStatus` function only. As we are checking for it every 
time we're updating the status.